azavea/climate-change-lab
# Feature/add tests

## Overview

Fixes test and linter runners, adds some tests, and fixes linter warnings.

### Notes

Upgraded typescript to be able to upgrade linter package versions.
Switches back to Chrome karma launcher, as PhantomJS is non-functional, due to read-only error stack.
Due to the Chrome karma launcher, a browser window briefly flashes open during test run; apparently this cannot be disabled (see [this issue](karma-runner/karma-chrome-launcher#27)).

Code changes outside of configuration and added tests were all made to address linter warnings... there were a lot.

## Testing Instructions

 * `npm start` should run successfully, to output of 'OK' message (no typescript check issues)
 * `npm run lint` should exit successfully, with no warnings (if there are warnings, npm exits in error)
 * UI should still function as before
 * `npm test` should run seven tests successfully
